src/t_hash.c

```diff
@@ -1053,6 +1053,59 @@ void hdelCommand(client *c) {
     addReplyLongLong(c, deleted);
 }
 
+void hgetdelCommand(client *c) {
+    int fields_index = 4;
+    int i, deleted = 0;
+    long long num_fields = 0;
+    bool keyremoved = false;
+
+    if (getLongLongFromObjectOrReply(c, c->argv[fields_index - 1], &num_fields, NULL) != C_OK) return;
+
+    /* Check that the parsed fields number matches the real provided number of fields */
+    if (!num_fields || num_fields != (c->argc - fields_index)) {
+        addReplyErrorObject(c, shared.syntaxerr);
```

Comment:
in other commands accepting fields we emit error (maybe we should do the same?):
```suggestion
        addReplyError(c, "numfields should be greater than 0 and match the provided number of fields");
```